### PR TITLE
feat(digital-twin-sample)added digital twin android example

### PR DESCRIPTION
**Pull request by davilu**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: digital-twin/Samples/device/AndroidSample/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        sourceCompatibility 1.8
+        targetCompatibility 1.8
+    }
+    compileSdkVersion 29
+    buildToolsVersion "29.0.2"
+    defaultConfig {
+        applicationId "com.microsoft.azure.sdk.iot.digitaltwin.sample"
+        minSdkVersion 19
+        targetSdkVersion 29
+        versionCode 1
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/device/AndroidSample/app/build.gradle]: Can we specify a wild card for the latest android API as oppose to hard-coding it with 29 as target?

File: digital-twin/Samples/device/AndroidSample/app/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/sample/EnvironmentalSensor.java [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+import static com.microsoft.azure.sdk.iot.digitaltwin.device.serializer.JsonSerializer.deserialize;
+import static com.microsoft.azure.sdk.iot.digitaltwin.device.serializer.JsonSerializer.serialize;
+import static java.util.Collections.singletonList;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+@Slf4j
+public class EnvironmentalSensor extends AbstractDigitalTwinInterfaceClient {
+    static final String ENVIRONMENTAL_SENSOR_INTERFACE_ID = "urn:csharp_sdk_sample:EnvironmentalSensor:1";
+    private static final String COMMAND_NOT_HANDLED_MESSAGE_PATTERN = "\"Command[%s] is not handled for interface[%s].\"";
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/device/AndroidSample/app/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/sample/EnvironmentalSensor.java]: urn contains csharp ? 🔢

File: digital-twin/Samples/device/JdkSample/pom.xml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>${lombok-version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.microsoft.azure.sdk.iot</groupId>
+            <artifactId>digital-twin-device-client-preview</artifactId>
+            <version>${digital-twin-device-client-version}</version>
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/device/JdkSample/pom.xml]: why cant you specify "LATEST"here as well ?

File: digital-twin/Samples/device/JdkSample/readme.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+This project contains a sample implementation of a simulated environmental sensor. It shows how to:
+
+  * Implement the environmental sensor interface
+  * Create an interfaceInstance for this interface
+  * Use the digital twin device client to register this interfaceInstance and interact with the Digital Twins services.
+  * How to respond to command invocations
+  * How to respond to property updates
+  
+For Android, please see [here](../AndroidSample)
+## How to run the sample
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/device/JdkSample/readme.md]: Broken link for Android

File: digital-twin/Samples/device/JdkSample/readme.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+In order to run this sample, you will need an IoT Hub. You will also need at least one device registered in this hub so that the sample can register as that device
+* [Setup Your IoT Hub][lnk-setup-iot-hub]
+
+### Run the sample
+
+From the base digital-twin folder in this repo, run the following command:
+
+```sh
+mvn clean install -DskipTests
+```
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/device/JdkSample/readme.md]: do you need -DskipTests for sample? Can you remove?

File: digital-twin/Samples/service/JdkSample/get-digital-twin/pom.xml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <version>${maven-jar-plugin-version}</version>
+                <configuration>
+                    <archive>
+                        <manifest>
+                            <addClasspath>true</addClasspath>
+                            <mainClass>com.microsoft.azure.sdk.iot.digitaltwin.sample.GetDigitalTwinSample</mainClass>
+                        </manifest>
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/get-digital-twin/pom.xml]: nit : follow correct tabs

File: digital-twin/Samples/service/JdkSample/get-digital-twin/readme.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+In order to run this sample, you will need an IoT Hub. You will also need at least one device registered in this hub so that the sample can lookup the state of that device
+* [Setup Your IoT Hub][lnk-setup-iot-hub]
+
+### Run the sample
+
+From the base digital-twin folder in this repo, run the following command:
+
+```sh
+mvn clean install -DskipTests
+```
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/get-digital-twin/readme.md]: do you need "-DskipTests" for sample run?

File: digital-twin/Samples/service/JdkSample/get-model/readme.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+In order to run this sample, you will need an IoT Hub. You will also need at least one device registered in this hub so that the sample can lookup the state of that device
+* [Setup Your IoT Hub][lnk-setup-iot-hub]
+
+### Run the sample
+
+From the base digital-twin folder in this repo, run the following command:
+
+```sh
+mvn clean install -DskipTests
+```
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/get-model/readme.md]: -DskipTests is not needed.

File: digital-twin/Samples/service/JdkSample/invoke-digital-twin-async-command/readme.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+### Other Prerequisites
+In order to run this sample, you will need an IoT Hub. You will also need at least one device registered in this hub so that the sample can invoke a command on that device
+* [Setup Your IoT Hub][lnk-setup-iot-hub]
+
+This device must also be registered to implement the interface whose async command you want to invoke, and the device
+must be online to receive the command invocation.
+
+To see sample code that demonstrates how to register a device to implement an interface and to listen for commands, see the device sample [here](../device-samples)
+
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/invoke-digital-twin-async-command/readme.md]: 404 here

File: digital-twin/Samples/service/JdkSample/invoke-digital-twin-async-command/readme.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+### Run the sample
+
+From the base digital-twin folder in this repo, run the following command:
+
+```sh
+mvn clean install -DskipTests
+```
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/invoke-digital-twin-async-command/readme.md]: -DskipTests is not needed

File: digital-twin/Samples/service/JdkSample/invoke-digital-twin-command/readme.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+### Other Prerequisites
+In order to run this sample, you will need an IoT Hub. You will also need at least one device registered in this hub so that the sample can invoke a command on that device
+* [Setup Your IoT Hub][lnk-setup-iot-hub]
+
+This device must also be registered to implement the interface whose command you want to invoke, and the device
+must be online to receive the command invocation.
+
+To see sample code that demonstrates how to register a device to implement an interface and to listen for commands, see the device sample [here](../device-samples)
+
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/invoke-digital-twin-command/readme.md]: 404 here

File: digital-twin/Samples/service/JdkSample/invoke-digital-twin-command/readme.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+### Run the sample
+
+From the base digital-twin folder in this repo, run the following command:
+
+```sh
+mvn clean install -DskipTests
+```
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/invoke-digital-twin-command/readme.md]: -DskipTests is not needed

File: digital-twin/Samples/service/JdkSample/pom.xml [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+        <module>update-digital-twin</module>
+        <module>get-model</module>
+        <module>invoke-digital-twin-command</module>
+        <module>invoke-digital-twin-async-command</module>
+    </modules>
+    <properties>
+		<java-sdk-device-version>1.7</java-sdk-device-version>
+        <java-sdk-service-version>1.8</java-sdk-service-version>
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <lombok-version>1.18.4</lombok-version>
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/pom.xml]: nit: please use proper tab

File: digital-twin/Samples/service/JdkSample/pom.xml [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+        <maven-compiler-plugin-version>3.8.1</maven-compiler-plugin-version>
+        <maven-source-plugin-version>3.1.0</maven-source-plugin-version>
+        <maven-shade-version>2.4</maven-shade-version>
+        <maven-jar-plugin-version>3.1.0</maven-jar-plugin-version>
+        <build-helper-version>3.0.0</build-helper-version>
+        <slf4j-version>1.7.25</slf4j-version>
+		<ms-rest-client-version>1.6.15</ms-rest-client-version>
+        <azure-eventhubs-version>3.0.0</azure-eventhubs-version>
+        <project-service-samples-version>1.0.0</project-service-samples-version>
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/pom.xml]: Do you need event hubs in samples?

File: digital-twin/Samples/service/JdkSample/pom.xml [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-log4j12</artifactId>
+            <version>${slf4j-version}</version>
+        </dependency>
+		<dependency>
+            <groupId>com.microsoft.rest</groupId>
+            <artifactId>client-runtime</artifactId>
+            <version>${ms-rest-client-version}</version>
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/pom.xml]: Do you need this for samples?

File: digital-twin/Samples/service/JdkSample/readme.md [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+
+[GetDigitalTwin](get-digital-twin) - Demonstrates how to check the state of a given digital twin
+
+[GetModel](get-model) - Demonstrates how to lookup the model definition for a given device capability model id, or interface id
+
+[InvokeCommand](invoke-digital-twin-command) - Demonstrates how to invoke a digital twin sync command on an active digital twin
+
+[InvokeAsyncCommand](invoke-digital-twin-async-command) - Demonstrates how to invoke a digital twin async command on an active digital twin
+
+[UpdateProperty](update-digital-twin) - Demonstrates how to update a single property on a digital twin
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/readme.md]: Please updates all links in this page. they result in 404

File: digital-twin/Samples/service/JdkSample/update-digital-twin/pom.xml [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <version>${maven-jar-plugin-version}</version>
+                <configuration>
+                    <archive>
+                        <manifest>
+                            <addClasspath>true</addClasspath>
+                            <mainClass>com.microsoft.azure.sdk.iot.digitaltwin.sample.UpdateDigitalTwinSample</mainClass>
+                        </manifest>
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/update-digital-twin/pom.xml]: nit:please use proper tab

File: digital-twin/Samples/service/JdkSample/update-digital-twin/readme.md [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+
+### Other Prerequisites
+In order to run this sample, you will need an IoT Hub. You will also need at least one device registered in this hub so that the sample can update properties on that device
+* [Setup Your IoT Hub][lnk-setup-iot-hub]
+
+While not mandatory, it is recommended that this registered device is online when the service client updates the property so the device can get a callback when its property is updated.
+The property update will still work when the device is offline, though.
+
+To see sample code that demonstrates how to register a device to implement an interface, see the device sample [here](../device-samples)
+
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/update-digital-twin/readme.md]: 404

File: digital-twin/Samples/service/JdkSample/update-digital-twin/readme.md [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+### Run the sample
+
+From the base digital-twin folder in this repo, run the following command:
+
+```sh
+mvn clean install -DskipTests
+```
```
Comment (prmathur-microsoft) [inline digital-twin/Samples/service/JdkSample/update-digital-twin/readme.md]: Do you need -DskipTests

File: digital-twin/Samples/device/AndroidSample/app/build.gradle (hunk 1 above)
Comment (davilu) [reply digital-twin/Samples/device/AndroidSample/app/build.gradle]: Not a good idea.

File: digital-twin/Samples/device/JdkSample/pom.xml (hunk 3 above)
Comment (davilu) [reply digital-twin/Samples/device/JdkSample/pom.xml]: The value is `LATEST`

File: digital-twin/Samples/device/JdkSample/readme.md (hunk 4 above)
Comment (davilu) [reply digital-twin/Samples/device/JdkSample/readme.md]: It works to me.

File: digital-twin/Samples/service/JdkSample/pom.xml (hunk 14 above)
Comment (davilu) [reply digital-twin/Samples/service/JdkSample/pom.xml]: Yes

File: digital-twin/Samples/service/JdkSample/pom.xml (hunk 15 above)
Comment (davilu) [reply digital-twin/Samples/service/JdkSample/pom.xml]: Yes

File: digital-twin/Samples/service/JdkSample/readme.md (hunk 16 above)
Comment (davilu) [reply digital-twin/Samples/service/JdkSample/readme.md]: It works to me.

File: digital-twin/Samples/device/AndroidSample/app/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/sample/EnvironmentalSensor.java (hunk 2 above)
Comment (davilu) [reply digital-twin/Samples/device/AndroidSample/app/src/main/java/com/microsoft/azure/sdk/iot/digitaltwin/sample/EnvironmentalSensor.java]: Fixed

File: digital-twin/Samples/device/JdkSample/readme.md (hunk 5 above)
Comment (davilu) [reply digital-twin/Samples/device/JdkSample/readme.md]: Removed

File: digital-twin/Samples/service/JdkSample/get-digital-twin/readme.md (hunk 7 above)
Comment (davilu) [reply digital-twin/Samples/service/JdkSample/get-digital-twin/readme.md]: removed

File: digital-twin/Samples/service/JdkSample/get-model/readme.md (hunk 8 above)
Comment (davilu) [reply digital-twin/Samples/service/JdkSample/get-model/readme.md]: removed

File: digital-twin/Samples/service/JdkSample/invoke-digital-twin-async-command/readme.md (hunk 9 above)
Comment (davilu) [reply digital-twin/Samples/service/JdkSample/invoke-digital-twin-async-command/readme.md]: fixed

File: digital-twin/Samples/service/JdkSample/invoke-digital-twin-async-command/readme.md (hunk 10 above)
Comment (davilu) [reply digital-twin/Samples/service/JdkSample/invoke-digital-twin-async-command/readme.md]: removed

File: digital-twin/Samples/service/JdkSample/invoke-digital-twin-command/readme.md (hunk 11 above)
Comment (davilu) [reply digital-twin/Samples/service/JdkSample/invoke-digital-twin-command/readme.md]: fixed

File: digital-twin/Samples/service/JdkSample/invoke-digital-twin-command/readme.md (hunk 12 above)
Comment (davilu) [reply digital-twin/Samples/service/JdkSample/invoke-digital-twin-command/readme.md]: removed

File: digital-twin/Samples/service/JdkSample/update-digital-twin/readme.md (hunk 18 above)
Comment (davilu) [reply digital-twin/Samples/service/JdkSample/update-digital-twin/readme.md]: fixed

File: digital-twin/Samples/service/JdkSample/update-digital-twin/readme.md (hunk 19 above)
Comment (davilu) [reply digital-twin/Samples/service/JdkSample/update-digital-twin/readme.md]: removed

File: digital-twin/Samples/device/JdkSample/readme.md (hunk 4 above)
Comment (prmathur-microsoft) [reply digital-twin/Samples/device/JdkSample/readme.md]: I can see it only after merge

File: digital-twin/Samples/service/JdkSample/pom.xml (hunk 15 above)
Comment (prmathur-microsoft) [reply digital-twin/Samples/service/JdkSample/pom.xml]: Samples wouldn't need it.

**Pull request merged**